examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C

```diff
@@ -0,0 +1,143 @@
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+// <h1>Vector Finite Element Example 1 - Solving an uncoupled Poisson Problem using DG</h1>
+// \author Alexander Lindsay
+// \date 2019
+//
+// This is the fifth vector FE example program.  It mimics the first example
+// but instead solves the uncoupled Poisson problem using a discontinuous basis (vector
+// monomial basis). Additionally it demonstrates use of a non-linear solver
+
+#include "libmesh/libmesh.h"
+#include "libmesh/distributed_mesh.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/nonlinear_implicit_system.h"
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/parameters.h"
+#include "libmesh/getpot.h"
+#include "libmesh/enum_solver_package.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/mesh_generation.h"
+#include "libmesh/enum_elem_type.h"
+
+namespace libMesh
+{
+template <typename>
+class NumericVector;
+template <typename>
+class SparseMatrix;
+}
+
+// Bring in everything from the libMesh namespace
+using namespace libMesh;
+
+// Function prototype for the exact solution.
+Real exact_solution(const int component, const Real x, const Real y, const Real z = 0.);
+
+// residual assembly function
+void compute_residual(const NumericVector<Number> & X,
+                     NumericVector<Number> & R,
+                     NonlinearImplicitSystem & S);
+
+// jacobian assembly function
+void compute_jacobian(const NumericVector<Number> & X,
+                     SparseMatrix<Number> & J,
+                     NonlinearImplicitSystem & S);
+
+int
+main(int argc, char ** argv)
+{
+  // Initialize libraries.
+  LibMeshInit init(argc, argv);
+
+  // This example requires a non-linear solver package.
+  libmesh_example_requires(libMesh::default_solver_package() == PETSC_SOLVERS ||
+                               libMesh::default_solver_package() == TRILINOS_SOLVERS,
+                           "--enable-petsc or --enable-trilinos");
+
+  // Parse input file
+  GetPot input_file("vector_fe_ex5.in");
+
+  // Read DG parameters
+  const Real epsilon = input_file("epsilon", -1);
+  const Real sigma = input_file("sigma", 6);
+
+  // Read mesh size
+  const std::size_t nx = input_file("nx", 15);
+  const std::size_t ny = input_file("ny", 15);
+
+  // Brief message to the user regarding the program name
+  // and command line arguments.
+  libMesh::out << "Running " << argv[0];
+
+  for (int i = 1; i < argc; i++)
+    libMesh::out << " " << argv[i];
+
+  libMesh::out << std::endl << std::endl;
+
+  // Skip this 2D example if libMesh was compiled as 1D-only.
+  libmesh_example_requires(2 <= LIBMESH_DIM, "2D support");
+
+  // Create a mesh, with dimension to be overridden later, on the
+  // default MPI communicator.
+  DistributedMesh mesh(init.comm());
```

Comment:
Do we want to force DistributedMesh here?

Reply:
I suppose not

Reply:
Changed to `Mesh`